x-pack/plugins/ml/server/models/data_frame_analytics/analytics_manager.ts

```diff
@@ -201,267 +208,379 @@ export class AnalyticsManager {
-          }
+  /**
+   * Prepares the initial elements for incoming jobId
+   * @param jobId
+   */
+  async getInitialElementsJobRoot(jobId: string): Promise<InitialElementsReturnType> {
+    const resultElements = [];
+    const modelElements = [];
+    const details: any = {};
+    const data = await this.getAnalyticsData(jobId);
+    const nextLinkId = data?.source?.index[0];
+    const nextType: JobMapNodeTypes = JOB_MAP_NODE_TYPES.INDEX;
+
+    const previousNodeId = `${data.id}-${JOB_MAP_NODE_TYPES.ANALYTICS}`;
+
+    resultElements.push({
+      data: {
+        id: previousNodeId,
+        label: data.id,
+        type: JOB_MAP_NODE_TYPES.ANALYTICS,
+        analysisType: getAnalysisType(data?.analysis),
+        isRoot: true,
+      },
+    });
 
-          if (link.meta?.created_by === 'data-frame-analytics') {
-            nextLinkId = link.meta.analytics;
-            nextType = JOB_MAP_NODE_TYPES.ANALYTICS;
-          }
+    details[previousNodeId] = data;
 
-          if (link.meta?.created_by === JOB_MAP_NODE_TYPES.TRANSFORM) {
-            nextLinkId = link.meta._transform?.transform;
-            nextType = JOB_MAP_NODE_TYPES.TRANSFORM;
-          }
-        } else if (isJobDataLinkReturnType(link) && link.isJob === true) {
-          data = link.jobData;
-          const nodeId = `${data.id}-${JOB_MAP_NODE_TYPES.ANALYTICS}`;
-          previousNodeId = nodeId;
+    const { modelElement, modelDetails, edgeElement } = this.getAnalyticsModelElements(jobId);
+    if (isAnalyticsMapNodeElement(modelElement)) {
+      modelElements.push(modelElement);
+      details[modelElement.data.id] = modelDetails;
+    }
+    if (isAnalyticsMapEdgeElement(edgeElement)) {
+      modelElements.push(edgeElement);
+    }
 
-          result.elements.unshift({
-            data: {
-              id: nodeId,
-              label: data.id,
-              type: JOB_MAP_NODE_TYPES.ANALYTICS,
-              analysisType: getAnalysisType(data?.analysis),
-            },
-          });
-          result.details[nodeId] = data;
-          nextLinkId = data?.source?.index[0];
-          nextType = JOB_MAP_NODE_TYPES.INDEX;
-
-          // Get inference model for analytics job and create model node
-          ({ modelElement, modelDetails, edgeElement } = this.getAnalyticsModelElements(data.id));
-          if (isAnalyticsMapNodeElement(modelElement)) {
-            modelElements.push(modelElement);
-            result.details[modelElement.data.id] = modelDetails;
-          }
-          if (isAnalyticsMapEdgeElement(edgeElement)) {
-            modelElements.push(edgeElement);
+    return { data, details, resultElements, modelElements, nextLinkId, nextType, previousNodeId };
+  }
+
+  /**
+   * Works backward from jobId or modelId to return related jobs, indices, models, and transforms
+   * @param jobId (optional)
+   * @param modelId (optional)
+   */
+  async getAnalyticsMap({
+    analyticsId,
+    modelId,
+  }: {
+    analyticsId?: string;
```

Comment:
Are we ever expecting both `analyticsId` and `modelId` to both be `undefined`? If we need/expect either or of this to be defined, I think it would be good to update the type.
```
 { analyticsId, modelId }: { analyticsId?: string; modelId?: string }
```

Reply:
Yes, we only expect one or the other. Updated type in 0e95f8993b8184dfdf3f02eaad3b476eda9ac79a